### ISSUE TITLE
org-mode reader does not support #+LABEL for figures and tables

### ISSUE THREAD

**Issue by sebhahn** [jgm/pandoc#2496]
First I thought the issue is coming from ox-pandoc, an Emacs plugin for pandoc (I've posted my issue here https://github.com/kawabata/ox-pandoc/issues/12#issuecomment-154106856)

However, it seems that this problem is caused by pandoc. I'm using pandoc 1.15.1.1, pandoc-citeproc 0.8.0.1, emacs 24.5.1 and org-mode 8.3.2

e.g.
`#+CAPTION: insert figure caption here`
`#+NAME: fig:1`
`#+LABEL: fig:1`
`[[img.png]]`

does create

`\begin{figure}[htbp]`
`\centering`
`\includegraphics{img.png}`
`\caption{insert figure caption here}`
`\end{figure}`

and the reference `[@fig:1]` gets correctly translated to `fig.~\ref{fig:1}` (using pandoc-citeproc), but it doesn't work because `\label{fig:1}` is missing in the Latex output? Somehow `#+NAME` needs to be provided as well, otherwise I'll get an error message from pandoc-citeproc.

Exporting with standard org-mode export to latex gives:

`\caption{\label{fig:orgparagraph1} insert figure caption here}`

still not the right label, but at least it is in the latex code

**Comment by tarleb** [comment]
Pandocs handling of figures is limited, labeled figures are unsupported atm. [This patch](https://github.com/tarleb/pandoc/commit/d2a5aee388b1042899167c45f9adcf35ffec1f0d) is probably the closest one can get, even though I'm not entirely sure if it does what you need.

**Comment by sebhahn** [comment]
The result should look like

`\begin{figure}[htbp]`
`\centering`
`\includegraphics{img.png}`
`\caption{insert figure caption here}`
`\label{fig:1}`
`\end{figure}`

**Comment by tarleb** [comment]
The problem is that this depends on issues  #813 and #1299.  People are working on those issues, but their resolution requires substantial changes in the underlying data structures and writers.  It's a complex and laborious task still pending completion.  As a result, there is currently no sane way to get proper figure references.  Sorry.

**Comment by sebhahn** [comment]
I'm looking forward to these new features :smile:

**Comment by tarleb** [comment]
Closing in favor of #813.
